eth_defi/gmx/contracts.py

```diff
@@ -0,0 +1,299 @@
+EXCHANGE_ROUTER_ABI: list = _load_abi("ExchangeRouter.json")
+ORACLE_ABI: list = _load_abi("Oracle.json")
+GLV_READER_ABI: list = _load_abi("GlvReader.json")
+BALANCE_ABI: list = _load_abi("balance.json")
+
+# TODO: Replace this to fetch the addresses dynamically from https://raw.githubusercontent.com/gmx-io/gmx-synthetics/refs/heads/v2.2-branch/docs/contracts.json
+# Token addresses by network
+NETWORK_TOKENS = {
+    "arbitrum": {"WETH": "0x82aF49447D8a07e3bd95BD0d56f35241523fBab1", "WBTC": "0x2f2a2543B76A4166549F7aaB2e75Bef0aefC5B0f", "USDC": "0xaf88d065e77c8cC2239327C5EDb3A432268e5831", "USDT": "0xFd086bC7CD5C481DCC9C85ebE478A1C0b69FCbb9", "ARB": "0x912CE59144191C1204E64559FE8253a0e49E6548", "LINK": "0xf97f4df75117a78c1A5a0DBb814Af92458539FB4", "wstETH": "0x5979D7b546E38E414F7E9822514be443A4800529"},
+    "avalanche": {"WAVAX": "0xB31f66AA3C1e785363F0875A1B74E27b85FD66c7", "WETH": "0x49D5c2BdFfac6CE2BFdB6640F4F80f226bc10bAB", "WBTC": "0x50b7545627a5162F82A992c33b87aDc75187B218", "USDC": "0xB97EF9Ef8734C71904D8002F8b6Bc66Dd9c48a6E", "USDT": "0x9702230A8Ea53601f5cD2dc00fDBc13d4dF4A8c7"},
+}
+
+
+def get_contract_addresses(chain: str) -> ContractAddresses:
+    """
+    Get GMX contract addresses for a specific network.
+
+    :param chain: Network name ("arbitrum" or "avalanche")
+    :return: Contract addresses for the network
+    :raises ValueError: If chain is not supported
+    """
+    # Handle the docstring keys in NETWORK_CONTRACTS
+    clean_contracts = {}
+    for key, value in NETWORK_CONTRACTS.items():
+        if isinstance(value, ContractAddresses):
+            # Handle case where chain name is embedded at end of docstring key
+            if key.endswith("arbitrum"):
+                clean_contracts["arbitrum"] = value
+            elif key.endswith("avalanche"):
+                clean_contracts["avalanche"] = value
+            else:
+                # Regular key
+                clean_contracts[key] = value
+
+    if chain not in clean_contracts:
+        raise ValueError(f"Unsupported chain: {chain}. Supported: {list(clean_contracts.keys())}")
+
+    return clean_contracts[chain]
+
+
+def get_reader_contract(web3: Web3, chain: str) -> Contract:
+    """
+    Get SyntheticsReader contract instance for a specific network.
+
+    :param web3: Web3 connection instance
+    :param chain: Network name
+    :return: Web3 contract instance for SyntheticsReader
+    """
+    addresses = get_contract_addresses(chain)
+    return web3.eth.contract(address=addresses.syntheticsreader, abi=READER_ABI)
+
+
+def get_datastore_contract(web3: Web3, chain: str) -> Contract:
+    """
+    Get DataStore contract instance for a specific network.
+
+    :param web3: Web3 connection instance
+    :param chain: Network name
+    :return: Web3 contract instance for DataStore
+    """
+    addresses = get_contract_addresses(chain)
+    return web3.eth.contract(address=addresses.datastore, abi=DATASTORE_ABI)
```

Comment:
Use `eth_defi.abi.get_contract` instead of `web3.eth.because it does caching of ABI files. 

Do not load and cache ABI on module import because it slows down import.

https://web3-ethereum-defi.readthedocs.io/api/core/_autosummary/eth_defi.abi.get_contract.html#eth_defi.abi.get_contract